src/test/java/me/karakelley/http/RequestTest.java

```diff
@@ -2,78 +2,113 @@
-    headers.put("Connection", "keep-alive");
+    Map<String, String> headers = setHeaders();
+    request = basicGetRequest();
 
-    request = new Request(HttpMethod.GET, "/", "HTTP/1.1", headers, "".getBytes(), 0);
     assertEquals(headers, request.getHeaders());
   }
 
   @Test
   void testOneHeader() {
-    HashMap headers = new HashMap();
-    headers.put("Host", "localhost:5000");
-    headers.put("Connection", "keep-alive");
-    request = new Request(null, null, null, headers, "".getBytes(), 0);
+    request = basicGetRequest();
+
     assertEquals("localhost:5000", request.getHeader("Host"));
   }
 
   @Test
   void testBadRequest() {
     try {
-      request = new Request(HttpMethod.GET, "/", null, null, "".getBytes(), 0);
-    } catch (Exception e) {
+      request = new Request.Builder().setMethod(HttpMethod.GET).setPath("/").setPort(0).build();
+      } catch (Exception e) {
       assertEquals("java.lang.ArrayIndexOutOfBoundsException: 2", e.getMessage());
     }
   }
 
   @Test
   void testGetHostAndPort() {
-    request = new Request(HttpMethod.GET, "/",  "HTTP/1.1", new HashMap<>(), "".getBytes(),  4000);
-    assertEquals(request.getHostAndPort(), "localhost:4000" );
+    request = basicGetRequest();
+    assertEquals(request.getHostAndPort(), "localhost:5000" );
   }
 
   @Test
   void testGetHostAndPortWithHostDefined() {
-    HashMap headers = new HashMap();
+    HashMap<String, String> headers = new HashMap<>();
     headers.put("Host", "test.com");
-    request = new Request(HttpMethod.GET, "/",  "HTTP/1.1", headers, "".getBytes(), 4000);
+    request = basicGetRequest(headers, 4000);
     assertEquals(request.getHostAndPort(), "test.com:4000" );
   }
 
   @Test
-  void testgetBody() throws IOException {
-    HashMap headers = new HashMap();
-    headers.put("Content-Length", 9);
-    Request request = new Request(HttpMethod.POST, "/test1.txt", "HTTP/1.1", headers, "txt=99999".getBytes(), 0);
+  void testGetBody() {
+    HashMap<String, String> headers = new HashMap<>();
+    headers.put("Content-Length", "9");
+    Request request = new Request.Builder()
+            .setMethod(HttpMethod.POST)
+            .setPath("/test1.txt")
+            .setProtocol("HTTP/1.1")
+            .setHeaders(headers)
+            .setBody("txt=99999".getBytes())
+            .setPort(0)
+            .build();
+
     assertEquals(new String(request.getBody()), "txt=99999");
   }
+
+  private Map<String, String> setHeaders() {
+    Map<String, String> headers = new HashMap<>();
+    headers.put("Host", "localhost:5000");
+    headers.put("Connection", "keep-alive");
+    return headers;
+  }
+
+  private Request basicGetRequest() {
+    return new Request.Builder()
+            .setMethod(HttpMethod.GET)
+            .setPath("/")
+            .setProtocol("HTTP/1.1")
+            .setHeaders(setHeaders())
+            .setPort(0)
+            .build();
+  }
+
+  private Request basicGetRequest(Map<String, String> headers, int port) {
```

Comment:
One pattern I've seen occasionally that I like is to call the same method in the body of the other-arity / overloaded version. (Poorly worded...) What I mean is, something like:
```
private Request basicGetRequest() {
  return basicGetRequest(setHeaders(), 0)
}
```